src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1090,27 +1087,11 @@ public string Scheme
         }
 
         //
-        //  Was the original string switched from m_String to m_OriginalUnicodeString
-        //  Will happen when Iri is turned on and we have unicode chars or of idn is
-        //  is on and we have an idn or unicode host.
+        //  Gets the exact string passed by a user.
+        //  The original string will switched from m_String to m_OriginalUnicodeString if
+        //  iri is turned on and we have non-ascii chars
         //
-        private bool OriginalStringSwitched
-        {
-            get
-            {
-                return (_iriParsing && InFact(Flags.HasUnicode));
-            }
-        }
-
-        //
-        //    Gets the exact string passed by a user.
-        public string OriginalString
-        {
-            get
-            {
-                return OriginalStringSwitched ? _originalUnicodeString : _string;
-            }
-        }
+        public string OriginalString => _originalUnicodeString ?? _string;
```

Comment:
There's no situation in which `_originalUnicodeString != null` is true but `(_iriParsing && InFact(Flags.HasUnicode))` is false?

Reply:
Yes, `_originalUnicodeString` is only set in [one place](https://github.com/dotnet/runtime/blob/dc62dd61071c239401ef6397a4d7ae29161d10a7/src/libraries/System.Private.Uri/src/System/UriExt.cs#L86-L93)
```c#
if (_iriParsing &&
    (CheckForUnicode(_string) || CheckForEscapedUnreserved(_string)))
{
    _flags |= Flags.HasUnicode;
    hasUnicode = true;
    // switch internal strings
    _originalUnicodeString = _string; // original string location changed
}
```